drivers/flash/Kconfig

```diff
@@ -23,6 +23,23 @@ config FLASH_HAS_PAGE_LAYOUT
 	  This option is enabled when the SoC flash driver supports
 	  retrieving the layout of flash memory pages.
 
+config FLASH_DEVICE_REQUIRES_ERASE
+	bool
+	help
+	  Device requires erase before write.
+
+config FLASH_DEVICE_HAS_ERASE
+	bool
+	default y if FLASH_DEVICE_REQUIRES_ERASE
+	help
+	  Device has erase function. It does not mean that erase is required
+	  by a device, it just means that it provides erase functionality.
+
+config FLASH_DEVICE_HAS_NO_ERASE
+	bool
+	help
+	  Device does not require erase.
+
```

Comment:
How will this support multiple flash devices in the same board having different behaviors?

Reply:
This is used to indicate what devices are on the board, the support depends on paths taken in code and checking run-time characteristic of a device called program-erase

Reply:
I'm ok with a config to help with codesize and such. But I think we need names that make that clearer. Something that suggests a general capability, not that sounds like it applies to a specific device. Drivers that need explicit erase, when it isn't defined shouldn't be configurable.